### PR TITLE
IGAPP-571: Upgrade to react-native v0.65.1

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Comment by klinzo** [comment]
- [ ]  Install Pods on iOS
- [ ] Test on iOS

**Comment by klinzo** [comment]
Currently blocked by react-native-reanmiated which does not support RN 0.65 right now.

**Review by Taggotty — APPROVED** [review]
Tested on Android emulator.
Would you mind to also do the [browserlist update](https://github.com/browserslist/browserslist#browsers-data-updating) here? :)

File: native/android/gradle.properties [hunk 1]
```diff
@@ -25,4 +25,4 @@ android.useAndroidX=true
 android.enableJetifier=true
 
 # Version of flipper SDK to use with React Native
-FLIPPER_VERSION=0.75.1
+FLIPPER_VERSION=0.93.0
```
[the remark this answers is not in the document]
Comment (klinzo) [reply native/android/gradle.properties:28]: Should be updated together with react native, see https://react-native-community.github.io/upgrade-helper/?from=0.64.2&to=0.65.1
I think sololoader will also show up there with a future react native version.

**Review by f1sh1918 — APPROVED** [review]
Tested on android & ios

File: native/android/build.gradle [hunk 2]
```diff
@@ -2,10 +2,10 @@
 
 buildscript {
     ext {
-        buildToolsVersion = "29.0.3"
+        buildToolsVersion = "30.0.2"
         minSdkVersion = 21
-        compileSdkVersion = 29
-        targetSdkVersion = 29
+        compileSdkVersion = 30
+        targetSdkVersion = 30
```
Comment (maxammann) [inline native/android/build.gradle:8]: I suppose all of these changes come from the offical diff, right?

File: native/android/build.gradle [hunk 3]
```diff
@@ -30,13 +30,15 @@ buildscript {
     }
     repositories {
         google()
+        mavenCentral()
+        // TODO Remove. Still required for some libraries.
```
Comment (maxammann) [inline native/android/build.gradle:34]: Why is this needed?

**Review by maxammann — APPROVED** [review]

File: native/android/build.gradle (hunk 3 above)
Comment (klinzo) [reply native/android/build.gradle:34]: See https://github.com/maplibre/maplibre-gl-native/issues/92, jcenter seems to be needed until mapbox-sdk-turf v5.12.0 while we are currently on v5.3.0. If I understood @f1sh1918 correctly we are not able to upgrade yet though, so we still need jcenter.

File: native/android/build.gradle (hunk 2 above)
Comment (klinzo) [reply native/android/build.gradle:8]: Yes, see https://react-native-community.github.io/upgrade-helper/?from=0.64.2&to=0.65.1
